### PR TITLE
[DEV-5856] Moving Elasticsearch ETL files around (part 2)

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
- Renaming management command `es_rapidloader` -> `elasticsearch_indexer`
    - GitHub shortcoming: doesn't correctly diff history from a `git mv` and instead treats the file as new
    - The original management command is unchanged by this PR other than imports
- Moved `es_etl_helpers.py` and `rapidloader.py` into new `elasticsearch_loader_helpers` directory

**Technical details:**
`es_rapidloader` is a passthrough management command to allow graceful switchover for ops scripts

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated (N/A)
3. [ ] Necessary PR reviewers:
    - [x] Backend
    - [ ] Operations
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed (N/A)
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket [DEV-5856](https://federal-spending-transparency.atlassian.net/browse/DEV-5856):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected Script (N/A)
    - [x] Before / After data comparison (N/A)

**Area for explaining above N/A when needed:**
```
Minor refactor to move files around
```

### CODE REVIEW

**Review by sethstoudenmier — APPROVED** [review]
Didn't test all options for "new" command but looks straightforward.

File: usaspending_api/etl/management/commands/elasticsearch_indexer.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        python3 manage.py es_rapidloader --index-name <NEW-INDEX-NAME> --create-new-index all
+
+        Running with --new-index will trigger several actions:
+        0. A view will be created in the source database for the ETL queries
+        1. A new index will be created from the value provided by --index-name (obviously)
+        2. A new index template will be loaded into the cluster to set mapping and index metadata
+        3. All aliases used by the API queries will be re-assigned to the new index
+        4. An alias for incremental indexes will be applied to the new index
+        5. If any previous indexes existed with the API aliases, they will be deleted.
+    """
```
Comment (kwhickey) [inline usaspending_api/etl/management/commands/elasticsearch_indexer.py:41]: Command name in docs needs updating. If this is out of date, probably best to review rest of docs to ensure it is accurately describing process / usage. You could move this to the django command help text instead.

File: usaspending_api/etl/management/commands/elasticsearch_indexer.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        loader = Rapidloader(config, elasticsearch_client)
+        loader.run_load_steps()
+        loader.complete_process()
+
+        logger.info(format_log("---------------------------------------------------------------"))
+        logger.info(format_log(f"Script completed in {perf_counter() - start:.2f}s"))
+        logger.info(format_log("---------------------------------------------------------------"))
+
+
+def process_cli_parameters(options: dict, es_client) -> dict:
```
Comment (kwhickey) [inline usaspending_api/etl/management/commands/elasticsearch_indexer.py:135]: more appropriate to name `process_cli_args` or `process_cli_arguments`

File: usaspending_api/etl/management/commands/elasticsearch_indexer.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            settings.ES_AWARDS_NAME_SUFFIX if config["load_type"] == "awards" else settings.ES_TRANSACTIONS_NAME_SUFFIX,
+        )
+    elif options["start_datetime"]:
+        config["starting_date"] = options["start_datetime"]
+    else:
+        # Due to the queries used for fetching postgres data,
+        #  `starting_date` needs to be present and a date before:
+        #      - The earliest records in S3.
+        #      - When all transaction records in the USAspending SQL database were updated.
+        #   And keep it timezone-award for S3
```
Comment (kwhickey) [inline usaspending_api/etl/management/commands/elasticsearch_indexer.py:168]: timezone-aware ?

File: usaspending_api/etl/management/commands/elasticsearch_indexer.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    }
+
+    config.update({k: v for k, v in arg_parse_options.items() if k in copy_args})
+    return config
+
+
+def fiscal_years_for_processing(options: list) -> list:
+    if "all" in options["fiscal_years"]:
+        return create_fiscal_year_list(start_year=parse_fiscal_year(settings.API_SEARCH_MIN_DATE))
+    return [int(x) for x in options["fiscal_years"]]
```
Comment (kwhickey) [inline usaspending_api/etl/management/commands/elasticsearch_indexer.py:229]: Something I noticed when you don't do the `all` FY option, is it (nicely) processes fiscal years in reverse. Typically the more relevant/hot data that people are looking for are going to be from the latest FY. Is this the place that would order the FYs to process, and if so can it be made to start with the latest FY and work backward?

If a failure happens late in the process, it increases the odds that the more-important data loaded before the failure.

**Review by kwhickey — COMMENTED** [review]
Minor naming and docs suggestions, plus consider ordering FYs in reverse.

File: usaspending_api/etl/management/commands/elasticsearch_indexer.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/etl/management/commands/elasticsearch_indexer.py:41]: Agree 100%. This PR was only to reorganize code with a few modifications as possible. The GH U/I did a poor job showing that.

File: usaspending_api/etl/management/commands/elasticsearch_indexer.py (hunk 4 above)
Comment (tony-sappe) [reply usaspending_api/etl/management/commands/elasticsearch_indexer.py:229]: Really? I thought the fiscal-years arg was required. Do you mean providing a list of FYs in descending order processes the ETL in the same order?

I expect future changes to DEV-5856 will significantly reduce the significance of fiscal years order in the processing.

**Pull request merged**